wren-ai-service/src/pipelines/sql_explanation/generation.py

```
@@ -0,0 +1,608 @@
+                                    "isFunctionCallOrMathematicalOperation": False,
+                                    "explanation": _extract_to_str(sql_explanation),
+                                },
+                            }
+                            for select_item, sql_explanation in zip(
+                                preprocessed_sql_analysis_results["selectItems"][
+                                    "withoutFunctionCallOrMathematicalOperation"
+                                ],
+                                sql_explanation_results["selectItems"][
+                                    "withoutFunctionCallOrMathematicalOperation"
+                                ],
+                            )
+                        ]
+
+                        results += sql_analysis_result_for_select_items
+                    elif preprocessed_sql_analysis_results.get(
+                        "sortings", []
+                    ) and sql_explanation_results.get("sortings", []):
+                        for (
+                            sorting,
+                            sql_explanation,
+                        ) in zip(
+                            preprocessed_sql_analysis_results["sortings"],
+                            sql_explanation_results["sortings"],
+                        ):
+                            results.append(
+                                {
+                                    "type": "sortings",
+                                    "payload": {
+                                        "id": sorting["id"],
+                                        "expression": sorting["values"],
+                                        "explanation": _extract_to_str(sql_explanation),
+                                    },
+                                }
+                            )
+        except Exception as e:
+            logger.exception(f"Error in GenerationPostProcessor: {e}")
```

Do we need to raise an exception after recording it in the log?

---

discussed offline, don't need to raise the error again.